pantheon/src/main/java/tech/pegasys/pantheon/cli/RLPSubCommand.java

```diff
@@ -0,0 +1,180 @@
+      mixinStandardHelpOptions = true)
+  static class EncodeSubCommand implements Runnable {
+
+    @SuppressWarnings("unused")
+    @ParentCommand
+    private RLPSubCommand parentCommand; // Picocli injects reference to parent command
+
+    @SuppressWarnings("unused")
+    @Spec
+    private CommandSpec spec;
+
+    @Option(
+        names = "--from",
+        paramLabel = MANDATORY_FILE_FORMAT_HELP,
+        description = "File containing JSON object to encode",
+        arity = "1..1")
+    private File jsonSourceFile = null;
+
+    @Option(
+        names = "--to",
+        paramLabel = MANDATORY_FILE_FORMAT_HELP,
+        description = "File to write encoded RPL string to.",
+        arity = "1..1")
+    private File rlpTargetFile = null;
+
+    @Override
+    public void run() {
+      checkNotNull(parentCommand);
+      readInput();
+    }
+
+    private void readInput() {
+      // if we have an output file defined, print to it
+      // otherwise print to defined output, usually standard output.
+      StringBuilder jsonData = new StringBuilder();
+
+      if (jsonSourceFile != null) {
+        try {
+          BufferedReader reader = Files.newBufferedReader(jsonSourceFile.toPath(), UTF_8);
+
+          String line;
+          while ((line = reader.readLine()) != null) jsonData.append(line);
+        } catch (IOException e) {
+          throw new ExecutionException(spec.commandLine(), "Unable to read JSON file.");
+        }
+      } else {
+        // get JSON data from standard input
+        try (Scanner scanner = new Scanner(parentCommand.in, UTF_8.name())) {
+          while (scanner.hasNextLine()) {
+            jsonData.append(String.join("", scanner.nextLine().split("\\s")));
+          }
+        }
+      }
+      encode(jsonData.toString());
+    }
+
+    private void encode(final String jsonInput) {
+      // map the json to the object matching the type option
+      // the object must be an RLPEncodable object
+      if (jsonInput == null || jsonInput.isEmpty()) {
+        throw new ExecutionException(
+            spec.commandLine(), "An error occurred while trying to read the JSON data.");
+      } else {
+        try {
+          JsonObject jsonObject = new JsonObject(jsonInput);
+          RLPEncodable objectToEncode = jsonObject.mapTo(IbftExtraData.class);
+          // encode and write the value
+          writeOutput(objectToEncode.encode());
+        } catch (DecodeException e) {
+          throw new ParameterException(
+              spec.commandLine(), "Unable to load the JSON data. Please check JSON input format.");
+        } catch (IllegalArgumentException e) {
+          throw new ParameterException(
+              spec.commandLine(),
+              "Unable to map the JSON data with IbftExtraData type. Please check JSON input format.");
+        }
+      }
+    }
+
+    private void writeOutput(final BytesValue rlpEncodedOutput) {
```

Comment:
should we also produce the bootnode information? Or is there another ticket to create that?

Reply:
It's just the genesys extra data part here. What's the bootnode information ?